charmcraft.yaml

```diff
@@ -23,3 +23,39 @@ parts:
       - libssl-dev
       - rustc
       - cargo
+
+name: gateway-api-integrator
+title: Gateway API Integrator
+description: |
+  A [Juju](https://juju.is/) [charm](https://juju.is/docs/olm/charmed-operators) deploying and managing external access to HTTP/HTTPS services in a
+  Kubernetes cluster via a Gateway and an HTTPRoute resource. This requires the Kubernetes
+  cluster in question to have an Gateway API controller already deployed into it.
+summary: An operator to manage gateway resources.
+links:
+  documentation: https://discourse.charmhub.io/t/
+  issues: https://github.com/canonical/gateway-api-integrator-operator/issues
+  source: https://github.com/canonical/gateway-api-integrator-operator
+  contact:
+    - https://launchpad.net/~canonical-is-devops
+assumes:
+  - k8s-api
+  - juju >= 3.3
+requires:
+  certificates:
+    interface: tls-certificates
+    limit: 1
+
+config:
+  options:
+    external-hostname:
+      default: ""
```

Comment:
is a default necessary?

Reply:
good point, I'm not sure if we need a default value here, I'll think about it and make a follow-up PR if needed